pkg/fuzzer/queue/retry.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package queue
+
+type retryer struct {
+	pq   *PlainQueue
+	base Source
+}
+
+// NewRetryer adds a layer that resends results with Status=Restarted.
+func NewRetryer(base Source) Source {
```

Comment:
Should other similar function also return Source and unexport their concrete type?

Reply:
In some cases, we do want to expose methods other than `Next()`. E.g. `PlainQueue` exposes interface both for the executor and for the source. For deduplication, we do want to have a method to dump the cache in order not to keep the requests in the memory forever.

Reply:
Ack. Let's switch to Source the ones that don't have additional methods. I see you already switched Callback and Order.